### DISCUSSION TITLE
[Idea] nicer docker-compose for developing

### DISCUSSION THREAD

**Discussion by tswfi** [PrestaShop/PrestaShop#31969] [Ideas]
### The problem

the current docker-compose.yml in the repo has room for improvements

* installed shop defaults to French language, English would be better suited for everybody to easily get up and running
* mysql uses version 5.. should be bumped to 8 (https://hub.docker.com/_/mysql/tags) or mariadb
* wait-for-it timeout is 60 seconds, that does not seem to be enough (and I do have a pretty fast machine, looks like the mysql startup is just slow for some reason)
* there is no reason to change dev mode off for this kind of development images
* startup process runs make assets on some old version of npm and it changes package-lock.json files
* something generates `config/defines.inc.phpe` file
* those two above mean that after running `docker compose up` you already have modified files in your repo you need to either ignore or checkout -- back
* documentation for "reset your dev environment" could be refined in the readme

And it could include some nice addons

* including mailcatcher would make testing emails locally easier

### The solution

I know that is quite a lot of things but I think the changes should still be pretty small and would benefit everybody in the long run.

### Do you plan to work on this subject?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by tswfi** [comment]
Testing the mysql change first. mysql:8 starts roughly four times faster than mysql:5

**Reply by tswfi** [reply]
as mysql starts faster the wait-for-it timeout can stay at 60 seconds

**Comment by tswfi** [comment]
```
sed -ie "s/define('_PS_MODE_DEV_', true);/define('_PS_MODE_DEV_',\ false);/g" /var/www/html/config/defines.inc.php 
```
in docker_run_git.sh looks wrong.

-i wants extension to be used for backup and uses the `e` it sees

should probably be:

`-i -e`

**Comment by tswfi** [comment]
And here is the pr: https://github.com/PrestaShop/PrestaShop/pull/31976

**Reply by tswfi** [reply]
Lets see if it CI process survives these changes :D

**Reply by tswfi** [reply]
Looks like it does :)

**Comment by tswfi** [comment]
Now mailcatcher would still be a nice addition.. with automatic configuration for smtp to point into it...

**Comment by eternoendless** [comment]
@jokesterfr what do you think?

**Comment by tswfi** [comment]
actually there is already a pr for it :) https://github.com/PrestaShop/PrestaShop/pull/32208

**Comment by kpodemski** [comment]
Hello @tswfi 

I'm sharing something that might be interesting for you:
https://github.com/PrestaShop/PrestaShop/pull/39305

**Reply by tswfi** [reply]
Thanks!

I think we can also close this discussion. Lots of things have already been improved since I started this thread :)
